### PR TITLE
Release: slate v2.1.8

### PR DESCRIPTION
**Pull request by themightychris**

### CODE REVIEW

File: php-classes/Slate/Progress/Narratives/Report.php [hunk 1]
```diff
@@ -71,4 +71,85 @@ class Report extends \VersionedRecord
...
+        $student = $this->Student;
+
+        if ($student->PrimaryEmailID && \Validators::email($student->Email))
+        {
+            // TODO: do we want the email address or recipient name?
+            $recipients[] = $student->EmailRecipient;
+            //$recipients[] = $student->Email;
+        }
+
+        $recipientTypes = explode(',' , $_REQUEST['Recipients']);
```
Comment (themightychris) [inline php-classes/Slate/Progress/Narratives/Report.php]: @BillClinton what's this for? Something is wrong if you're reading `$_REQUEST` from a model

File: php-classes/Slate/Progress/Narratives/Report.php [hunk 2]
```diff
@@ -71,4 +71,85 @@ class Report extends \VersionedRecord
...
+                $recipients[] = $student->Advisor->EmailRecipient;
+                //$recipients[] = $student->Advisor->Email;
+            }
+        }
+
+        if (in_array('Parents',$recipientTypes)) {
+
+            $guardianRelationships = Relationship::getAllByWhere(array(
+                'PersonID' => $student->ID
+                ,'Class' => 'Emergence\\People\\GuardianRelationship'
```
Comment (themightychris) [inline php-classes/Slate/Progress/Narratives/Report.php]: @BillClinton always use the new `::class` syntax when you need a fully-qualified class name string

File: php-classes/Slate/Progress/Narratives/Report.php [hunk 3]
```diff
@@ -71,4 +71,85 @@ class Report extends \VersionedRecord
...
+            ));
+
+            foreach($guardianRelationships as $guardianRelationship)
+            {
+                $relatedPerson = $guardianRelationship->RelatedPerson;
+                if ($relatedPerson->PrimaryEmailID && \Validators::email($relatedPerson->Email))
+                {
```
Comment (themightychris) [inline php-classes/Slate/Progress/Narratives/Report.php]: @BillClinton follow PSR style guidelines consistently for spacing/linebreaks around control structures, code not conforming will no longer be merged

File: php-classes/Slate/Progress/Narratives/Report.php [hunk 4]
```diff
@@ -71,4 +71,85 @@ class Report extends \VersionedRecord
...
         ]
     ];
-}
\ No newline at end of file
+
+    public static $searchConditions = array(
+        'narrativeID' => array(
+            'qualifiers' => array('narrativeid')
+            ,'points' => 2
+            ,'sql' => 'ID=%u'
```
Comment (themightychris) [inline php-classes/Slate/Progress/Narratives/Report.php]: @BillClinton leading commas aren't used anymore and this file was already cleaned up

File: php-classes/Slate/Progress/Narratives/Report.php [hunk 5]
```diff
@@ -71,4 +71,85 @@ class Report extends \VersionedRecord
             'class' => \Slate\Term::class
         ]
     ];
-}
\ No newline at end of file
+
+    public static $searchConditions = array(
```
Comment (themightychris) [inline php-classes/Slate/Progress/Narratives/Report.php]: @BillClinton the new array syntax `[ .. ]` is to be used on all new code

**Comment by themightychris** [comment]
@BillClinton will you have the chance any time soon to clean up the issues I noted in this PR so far? If not I can take care of them but let me know that you've reviewed them at least for next time

File: php-classes/Slate/Progress/Narratives/Report.php (hunk 2 above)
Comment (BillClinton) [inline php-classes/Slate/Progress/Narratives/Report.php]: @themightychris corrected in 4feb819

File: php-classes/Slate/Progress/Narratives/Report.php (hunk 5 above)
Comment (BillClinton) [inline php-classes/Slate/Progress/Narratives/Report.php]: @themightychris corrected in 4ca7156

File: php-classes/Slate/Progress/Narratives/Report.php (hunk 4 above)
Comment (BillClinton) [inline php-classes/Slate/Progress/Narratives/Report.php]: @themightychris corrected in 4ca7156

File: php-classes/Slate/Progress/Narratives/Report.php (hunk 3 above)
Comment (BillClinton) [inline php-classes/Slate/Progress/Narratives/Report.php]: @themightychris corrected in 4ca7156

File: php-classes/Slate/Progress/Narratives/Report.php [hunk 6]
```diff
@@ -71,4 +64,100 @@ class Report extends \VersionedRecord
             'class' => \Slate\Term::class
         ]
     ];
-}
\ No newline at end of file
+
+    public static $searchConditions = [
+        'narrativeID' => [
```
Comment (themightychris) [inline php-classes/Slate/Progress/Narratives/Report.php]: @BillClinton these are typically keyed after the field they search, capitalizing them like `narrativeID` isn't consistent with anywhere else, so change that to `NarrativeID`

For the qualifiers the pattern is to use hyphens as word separator like good does, no `narrativeid:123` -> `narrative-id:123`

**Pull request merged**